src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -142,7 +143,23 @@ class Gemini(session: SparkSession, log: Slf4jLogger, keyspace: String = Gemini.
   }
 }
 
-case class RepoFile(repo: String, file: String, sha: String)
+object URLFormatter {
+  val services = Map(
+    "github.com" ->  "https://%s/blob/%s/%s",
+    "bitbucket.org" -> "https://%s/src/%s/%s",
+    "gitlab.com" -> "https://%s/blob/%s/%s"
+  )
+  val default = ("", "repo: %s ref_hash: %s file: %s")
+
+  def format(repo: String, ref_hash: String, file: String): String = {
+    services.find{case (h, _) => repo.startsWith(h)}
+      .getOrElse(default)._2.format(repo.replaceFirst("\\.git$", ""), ref_hash, file)
+  }
```

Comment:
Probably a nitpick, but method chaining might assist readability and is kind of customary to have whitespace around `{` like in http://twitter.github.io/effectivescala/#Collections-Style

Also, outer `{}` can be skipped in this case as well